controllers/webhook/upstream_services.go

```diff
@@ -1,114 +1,90 @@
-	DataContentType string                    `json:"datacontenttype"`
-	Data            common.RunCompletionEvent `json:"data"`
-}
-
-func NewArgoEventBody(data common.RunCompletionEvent) ArgoEventBody {
-	return ArgoEventBody{
-		Specversion:     "1.0",
-		Id:              strconv.FormatInt(time.Now().UnixNano(), 10),
-		Source:          "argo-event-webhook",
-		DataContentType: HttpContentTypeJSON,
-		Data:            data,
-	}
-}
-
-func (hw ArgoEventWebhook) call(ctx context.Context, ed EventData) error {
-	logger := common.LoggerFromContext(ctx)
-	requestBody := NewArgoEventBody(ed.RunCompletionEvent)
-	bodyBytes, err := json.Marshal(requestBody)
+func (gnt GrpcNatsTrigger) call(ctx context.Context, event common.RunCompletionEvent) error {
+	runCompletionEvent, err := RunCompletionEventToProto(event)
 	if err != nil {
 		return err
 	}
+	_, err = gnt.Client.ProcessEventFeed(ctx, runCompletionEvent)
+	return err
+}
 
-	req, err := hw.buildRequest(ctx, bodyBytes)
+func RunCompletionEventToProto(event common.RunCompletionEvent) (*pb.RunCompletionEvent, error) {
+	pipelineName, err := event.PipelineName.String()
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("failed to format pipeline name for proto run completion event: %w", err)
 	}
-	transferHeaders(ed.Header, req)
 
-	response, err := hw.Client.Do(req)
+	runConfigurationName, err := event.RunConfigurationName.String()
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("failed to format run configuration name for proto run completion event: %w", err)
 	}
-	defer func(Body io.ReadCloser) {
-		err := Body.Close()
-		if err != nil {
-			logger.Error(err, "Failed to close body")
-		}
-	}(response.Body)
 
-	// Fully consume the response body
-	_, err = io.Copy(io.Discard, response.Body)
+	runName, err := event.RunName.String()
 	if err != nil {
-		logger.Error(err, "Failed to fully consume response body")
+		return nil, fmt.Errorf("failed to format run name for proto run completion event: %w", err)
 	}
 
-	if response.StatusCode/100 != 2 {
-		return errors.New(fmt.Sprintf("Upstream service returned error, http status code: [%s]", response.Status))
+	var artifacts []*pb.ServingModelArtifact
+	for _, artifact := range event.ServingModelArtifacts {
+		artifacts = append(artifacts, &pb.ServingModelArtifact{
+			Location: artifact.Location,
+			Name:     artifact.Name,
+		})
 	}
-	return nil
-}
 
-func transferHeaders(headers http.Header, req *http.Request) {
-	for headerName, headerValues := range headers {
-		for _, headerValue := range headerValues {
-			req.Header.Add(headerName, headerValue)
-		}
+	runCompletionEvent := pb.RunCompletionEvent{
+		PipelineName:          pipelineName,
+		Provider:              event.Provider,
+		RunConfigurationName:  runConfigurationName,
+		RunId:                 event.RunId,
+		RunName:               runName,
+		ServingModelArtifacts: artifacts,
+		Status:                string(event.Status),
```

Comment:
why is rce treated differently from other strings which don't have what is effectively a string type alias

Reply:
status is of `type RunCompletionStatus string` type alias  means we need to convert it back to its primitive `string` type to satisfy the compiler.   

`Artifact.Name` is already a `string` as below.

```
type Artifact struct {
	Name     string `json:"name"`
	Location string `json:"location"`
}
```

Reply:
ok. But given the other [comment](https://github.com/sky-uk/kfp-operator/pull/377#discussion_r1808797425), I guess this doesn't really matter. Can revisit if it does.